zproject/urls.py

```diff
@@ -565,37 +565,11 @@
     url(r'^terms/$', zerver.views.portico.terms_view, name='terms'),
     url(r'^privacy/$', zerver.views.portico.privacy_view, name='privacy'),
 
-    url(r'^config-error/google$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'backend_error': True, 'social_backend_name': 'google'},),
-    url(r'^config-error/github$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'backend_error': True, 'social_backend_name': 'github'},),
-    url(r'^config-error/gitlab$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'backend_error': True, 'social_backend_name': 'gitlab'},),
-    url(r'^config-error/smtp$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'smtp_error': True},),
-    url(r'^config-error/ldap$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'ldap_error_realm_is_none': True},
-        name='ldap_error_realm_is_none'),
-    url(r'^config-error/dev$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'dev_not_supported_error': True},
-        name='dev_not_supported'),
-    url(r'^config-error/saml$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'saml_error': True},),
-    url(r'^config-error/remoteuser/backend_disabled$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'remoteuser_error_backend_disabled': True},),
-    url(r'^config-error/remoteuser/remote_user_header_missing$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'remoteuser_error_remote_user_header_missing': True},),
 ]
 
+for config_error_url in CONFIG_ERRORS:
+    i18n_urls.append(config_error_url.url_object())
```

Comment:
Hmm, couln't we have
``url(r'^config-error/(?P<error_category_name>[\w,-]+)$', ....``, handled by function ``f``, ``CONFIG_ERRORS`` be a dict mapping the names to the class:
CONFIG_ERRORS = {
    "google": BackendAuthConfigError('google', has_markdown_file=True),
}

and ``f`` would do something like
```
error_class = CONFIG_ERRORS[error_category_name]
TemplateView.as_view(
                   template_name=error_class.CONFIG_ERROR_PATH,
                   extra_context={'has_markdown_file': error_class.has_markdown_file,
                    'social_backend_name': error_class.social_backend_name,
                    'error_name': error_class.error_name
                    })
```

Since we can just pass content to ``as_view`` https://docs.djangoproject.com/en/3.0/ref/class-based-views/mixins-simple/#django.views.generic.base.ContextMixin.extra_context